File: src/1-Problem12.pas
program OperasiBilangan;
uses crt;

{ Kamus }
var
  a, b, c, d: integer;

{ Algoritma }
begin
  { Membaca masukan }
  write('Masukkan nilai A: '); readln(a);
  write('Masukkan nilai B: '); readln(b);
  write('Masukkan nilai C: '); readln(c);
  write('Masukkan nilai D: '); readln(d);
  
  { Memeriksa jawaban yang mungkin }
  write('Persamaan yang mungkin adalah: ');
  if (a+b = c+d) then
    writeln(a, ' + ', b, ' = ', c, ' + ', d)
  else if (a+b = c-d) then
    writeln(a, ' + ', b, ' = ', c, ' - ', d)
  else if (a+b = c*d) then
    writeln(a, ' + ', b, ' = ', c, ' * ', d)
  else if (a+b = c div d) then
    writeln(a, ' + ', b, ' = ', c, ' / ', d)

  else if (a-b = c+d) then
    writeln(a, ' - ', b, ' = ', c, ' + ', d)
  else if (a-b = c-d) then
    writeln(a, ' - ', b, ' = ', c, ' - ', d)
  else if (a-b = c*d) then
    writeln(a, ' - ', b, ' = ', c, ' * ', d)
  else if (a-b = c div d) then
    writeln(a, ' - ', b, ' = ', c, ' / ', d)

  else if (a*b = c+d) then
    writeln(a, ' * ', b, ' = ', c, ' + ', d)
  else if (a*b = c-d) then
    writeln(a, ' * ', b, ' = ', c, ' - ', d)
  else if (a*b = c*d) then
    writeln(a, ' * ', b, ' = ', c, ' * ', d)
  else if (a*b = c div d) then
    writeln(a, ' * ', b, ' = ', c, ' / ', d)

  else if (a div b = c+d) then
    writeln(a, ' / ', b, ' = ', c, ' + ', d)
  else if (a div b = c-d) then
    writeln(a, ' / ', b, ' = ', c, ' - ', d)
  else if (a div b = c*d) then
    writeln(a, ' / ', b, ' = ', c, ' * ', d)
  else if (a div b = c div d) then
    writeln(a, ' / ', b, ' = ', c, ' / ', d);

end.

File: 1-Problem12.py
a=input("Masukkan nilai A: ");
b=input("Masukkan nilai B: ");
c=input("Masukkan nilai C: ");
d=input("Masukkan nilai D: ");

A=int(a)
B=int(b)
C=int(c)
D=int(d)

if (D!=0):
	if((A+B)==int(C/D)):
		print ("Persamaan yang mungkin adalah: "+a+" + "+b+" = "+c+" / "+d)
	if((A-B)==int(C/D)):
		print ("Persamaan yang mungkin adalah: "+a+" - "+b+" = "+c+" / "+d)
	if((A*B)==int(C/D)):
		print ("Persamaan yang mungkin adalah: "+a+" * "+b+" = "+c+" / "+d)
if (B!=0):
	if(int(A/B)==(C+D)):
		print ("Persamaan yang mungkin adalah: "+a+" / "+b+" = "+c+" + "+d)
	if(int(A/B)==(C-D)):
		print ("Persamaan yang mungkin adalah: "+a+" / "+b+" = "+c+" - "+d)
	if(int(A/B)==(C*D)):
		print ("Persamaan yang mungkin adalah: "+a+" / "+b+" = "+c+" * "+d)
if ((B!=0) and (D!=0)):
	if(int(A/B)==int(C/D)):
		print ("Persamaan yang mungkin adalah: "+a+" / "+b+" = "+c+" / "+d)

if((A+B) == (C-D)):
	print ("Persamaan yang mungkin adalah: "+a+" + "+b+" = "+c+" - "+d)
if((A+B)==(C*D)):
	print ("Persamaan yang mungkin adalah: "+a+" + "+b+" = "+c+" * "+d)
if((A+B)==(C+D)):
	print ("Persamaan yang mungkin adalah: "+a+" + "+b+" = "+c+" + "+d)
if((A-B)==(C+D)):
	print ("Persamaan yang mungkin adalah: "+a+" - "+b+" = "+c+" + "+d)
if((A-B)==(C-D)):
	print ("Persamaan yang mungkin adalah: "+a+" - "+b+" = "+c+" - "+d)
if((A-B)==(C*D)):
	print ("Persamaan yang mungkin adalah: "+a+" - "+b+" = "+c+" * "+d)

if((A*B)==(C+D)):
	print ("Persamaan yang mungkin adalah: "+a+" * "+b+" = "+c+" + "+d)
if((A*B)==(C-D)):
	print ("Persamaan yang mungkin adalah: "+a+" * "+b+" = "+c+" - "+d)
if((A*B)==(C*D)):
	print ("Persamaan yang mungkin adalah: "+a+" * "+b+" = "+c+" * "+d)


